src/vs/workbench/browser/parts/editor/textDiffEditor.ts

```diff
@@ -60,6 +60,19 @@ export class TextDiffEditor extends BaseTextEditor implements ITextDiffEditorPan
 		this.doSaveOrClearTextDiffEditorViewState(editor);
 	}
 
+	invokeWithinContext<T>(fn: (accessor: ServicesAccessor) => T): T | null {
+		const control = this.getControl();
+		if (!control) {
+			return null;
+		}
+
+		if (control.getOriginalEditor().hasTextFocus()) {
+			return control.getOriginalEditor().invokeWithinContext(fn);
+		} else {
+			return control.getModifiedEditor().invokeWithinContext(fn);
+		}
```

Comment:
since you used ternary if condition in invokeWithinContext template implementation in textEditor.ts, it's possible to use the same style here as well: 
```suggestion
		return (control.getOriginalEditor().hasTextFocus())?
                      control.getOriginalEditor().invokeWithinContext(fn):
                      control.getModifiedEditor().invokeWithinContext(fn);
```